python/dgl/distributed/graph_partition_book.py

```diff
@@ -0,0 +1,250 @@
+            path of graph partition file.
+        ip_config_file : str
+            path of IP configuration file.
+        """
+        with open(partition_config_file) as conf_f:
+            self._part_meta = json.load(conf_f)
+        # Read ip list from ip_config_file
+        self._ip_list = []
+        lines = [line.rstrip('\n') for line in open(ip_config_file)]
+        # we assume that ip is sorted by machine ID in ip_config_file
+        for line in lines:
+            ip_addr, _, _ = line.split(' ')
+            self._ip_list.append(ip_addr)
+        # Get number of partitions
+        assert 'num_parts' in self._part_meta, "cannot get the number of partitions."
+        self._num_partitions = self._part_meta['num_parts']
+        assert self._num_partitions > 0, 'num_partitions cannot be a negative number.'
+        # Get part_files
+        self._part_files = []
+        for part_id in range(self._num_partitions):
+            assert 'part-{}'.format(part_id) in self._part_meta, \
+            "part-{} does not exist".format(part_id)
+            part_files = self._part_meta['part-{}'.format(part_id)]
+            self._part_files.append(part_files)
+        # Get nid2partid
+        assert 'node_map' in self._part_meta, "cannot get the node map."
+        self._nid2partid = F.tensor(np.load(self._part_meta['node_map']))
+        # Get eid2partid
+        assert 'edge_map' in self._part_meta, "cannot get the edge map."
+        self._eid2partid = F.tensor(np.load(self._part_meta['edge_map']))
+        # Get meta data
+        self._meta_data = []
+        _, nid_count = np.unique(F.asnumpy(self._nid2partid), return_counts=True)
+        _, eid_count = np.unique(F.asnumpy(self._eid2partid), return_counts=True)
+        for part_id in range(self._num_partitions):
+            part_info = {}
+            part_info['machine_id'] = part_id
+            part_info['ip'] = self._ip_list[part_id]
+            part_info['num_nodes'] = nid_count[part_id]
+            part_info['num_edges'] = eid_count[part_id]
+            self._meta_data.append(part_info)
+        # Get partid2nids
+        self._partid2nids = []
+        sorted_nid = F.tensor(np.argsort(F.asnumpy(self._nid2partid)))
+        start = 0
+        for offset in nid_count:
+            part_nids = sorted_nid[start:start+offset]
+            start += offset
+            self._partid2nids.append(part_nids)
+        # Get partid2eids
+        self._partid2eids = []
+        sorted_eid = F.tensor(np.argsort(F.asnumpy(self._eid2partid)))
+        start = 0
+        for offset in eid_count:
+            part_eids = sorted_eid[start:start+offset]
+            start += offset
+            self._partid2eids.append(part_eids)
+        # Get part_graphs
+        self._part_graphs = []
+        for part_id in range(self._num_partitions):
+            graph = load_graphs(self._part_files[part_id]['part_graph'])[0][0]
+            self._part_graphs.append(graph)
```

Comment:
i think we shouldn't load partition graph structures.

Reply:
+1